xbmc/music/MusicDatabase.cpp

```diff
@@ -7821,6 +7970,17 @@ void CMusicDatabase::UpdateTables(int version)
     // add strDiscSubtitles to song table
     m_pDS->exec("ALTER TABLE song ADD strDiscSubtitle TEXT \n");
   }
+  if (version < 74)
+  {
+    // add strOrigReleaseDate and strAlbumReleaseDate to album table
+    m_pDS->exec("ALTER TABLE album ADD strOrigReleaseDate TEXT");
+    m_pDS->exec("ALTER TABLE album Add strAlbumReleaseDate TEXT");
```

Comment:
No, it doesn't always have iYear values and this is actually something I only realized when I started on this.  Because of Picard's 'feature' of not populating the year field for flac files if it has a full date, some of my albums had no year.
But yes, populating from iYear on db upgrade would seem to be the sane thing to do although as iYear is retained anyway I don't think it will make any difference other than to the new listitems if users don't rescan.  The new sort by etc will just return '..' for an empty directory.